unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs

```diff
@@ -280,6 +347,20 @@ public void SetMinWidthForFlexibleItems(float minWidthForFlexibleItems)
         SetItemSize(model.itemSize);
     }
 
+    public void SetItems(BaseComponentView prefab, int amountOfItems)
+    {
+        RemoveItems();
+
+        for (int i = 0; i < amountOfItems; i++)
+        {
+            BaseComponentView instanciatedItem = Instantiate(prefab);
+            CreateItem(instanciatedItem, $"Item{i}");
+        }
+
+        SetItemSize(model.itemSize);
+    }
```

Comment:
Why we need this new `SetItems(BaseComponentView prefab, int amountOfItems)` method? Couldn't you call the `SetItems(List<BaseComponentView> items)` making the instantiations before? IMHO an UI Component of type "container" like this shouldn't have the reponsibility of instantiate its own items (imagine the case of a C# List: it offers you the "way" to manage a list of items, but the own items should be given by you as consumer of the list).

Reply:
I think that his way we can always introduce a pool manager in the container without the need to implement a custom one in each controller that uses this kind of component.

Android ended implementing the pool in the container since it will give the pool speed by default without the need to implement it each time you want to use a container

Reply:
Ok, let introduce the `SetItems()` from prefabs in our container UI components, but then please add it in the Carousel component too (it is also a container).